ui/src/app/services/workspace-storage.service.ts

```diff
@@ -0,0 +1,77 @@
+import 'rxjs/Rx';
+
+import {Injectable} from '@angular/core';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {Observable} from 'rxjs/Observable';
+import {ReplaySubject} from 'rxjs/ReplaySubject';
+
+
+import {Workspace} from 'generated';
+import {WorkspacesService} from 'generated';
+import {WorkspaceAccessLevel} from 'generated';
+
+export interface WorkspaceData extends Workspace {
+  accessLevel: WorkspaceAccessLevel;
+}
+
+export const BLANK_WORKSPACE: WorkspaceData = {
+  accessLevel: WorkspaceAccessLevel.NOACCESS,
+  name: '',
+  researchPurpose: {
+    diseaseFocusedResearch: false,
+    methodsDevelopment: false,
+    controlSet: false,
+    aggregateAnalysis: false,
+    ancestry: false,
+    commercialPurpose: false,
+    population: false,
+    reviewRequested: false,
+  }
+};
+
+@Injectable()
+export class WorkspaceStorageService {
+  private activeCall = false;
+  private activeWorkspace = new BehaviorSubject<WorkspaceData>(BLANK_WORKSPACE);
+  public activeWorkspace$ = this.activeWorkspace.asObservable();
```

Comment:
multiplexing this observable across multiple different workspaces is a bit scary. e.g. imagine a navigate from `workspaces/a` -> `workspaces/b`: `activeWorkspace$` doesn't necessarily track what's currently on the screen for the user. What will the user see during the navigate? If the navigate fails, things may be put into a bad state.

Or more practically with this PR, navigate from `workspace/a` -> `/workspaces` -> `workspace/b` - does the user see a flash of workspace `a` while `b` is loading?

Reply:
If the latency is high, they could. I can add some checks to the subscriptions to prevent that.